ffn/utils/proofreading.py

```diff
@@ -182,58 +260,293 @@ def __init__(self, objects, bad, num_to_prefetch=10, locations=None):
+    """
+    return "num_bad: %d" % len(self.bad)
 
-    self.update_batch()
+  def mark_bad(self) -> None:
+    """Mark an object or group of objects as bad.
 
-  def custom_msg(self):
-    return 'num_bad: %d' % len(self.bad)
-
-  def mark_bad(self):
+    If the batch size is greater than 1, the user is prompted to decrease
+    the batch size.
+    """
     if self.batch > 1:
-      self.update_msg('decrease batch to 1 to mark objects bad')
+      self.update_msg("decrease batch to 1 to mark objects bad")
       return
 
-    sids = self.todo[self.index]['seg']
+    sids = self.todo[self.index]["seg"]
     if len(sids) == 1:
       self.bad.add(list(sids)[0])
     else:
       self.bad.add(frozenset(sids))
 
-    self.update_msg('marked bad: %r' % (sids,))
+    self.update_msg("marked bad: %r" % (sids,))
     self.next_batch()
 
-  def mark_removed_bad(self):
+  def mark_removed_bad(self) -> None:
+    """From the set of original objects mark those bad that are not displayed.
+    Update the message with the IDs of the newly marked bad objects.
+    """
     original = set(self.list_segments())
-    new_bad = original - set(self.viewer.state.layers['seg'].segments)
+    new_bad = original - set(self.viewer.state.layers["seg"].segments)
     if new_bad:
       self.bad |= new_bad
-      self.update_msg('marked bad: %r' % (new_bad,))
+      self.update_msg("marked bad: %r" % (new_bad,))
+
+
+class ObjectReviewStoreLocation(ObjectReview):
+  """Class to mark and store locations of errors in the segmentation
+
+  To mark a merger, move the cursor to a spot of the false merger and press
+  'w'. Then, move the cursor to a spot within the object that should belong to
+  a separate object and press 'shift + W'. Yellow point annotations indicate
+  the merger. For split errors, proceed in similar manner but press 'd' and
+  'shift + D', which will display blue annotations.
+  Marked locations can be deleted either by pressing 'ctrl + Z' (to delete the
+  last marked location) or by hovering the cursor over one of the point
+  annotations and pressing 'ctrl + v'.
+
+  Attributes:
+    seg_error_coordinates: A mapping of annotation identifier substrings to
+                           error coordinate pairs.
+    Example: {'m0': [[x1,y1,z1],[x2,y2,z2]], 's0':[[x1,y1,z1],[x2,y2,z2]], ...}
+      - Keys starting with 'm' indicate merge errors.
+      - Keys starting with 's' indicate split errors.
+    temp_coord_list: Temporary storage for coordinates.
+  """
+
+  def __init__(
+      self,
+      objects: list,
+      bad: list,
+      seg_error_coordinates: Optional[list[str, list[list[int]]]] = {},
+      load_annotations: bool = False,
+  ) -> None:
+    """Initialize the ObjectReviewStoreLocation class.
+
+    Args:
+      objects: A list of objects.
+      bad: A list of bad objects or markers.
+      seg_error_coordinates: A dictionary of error coordinates.
+      load_annotations: A flag to indicate if annotations should be loaded.
```

Comment:
What's a use case where you would provide seg_error_coordinates, but set load_annotations to False?

Reply:
A combination of laziness and generak flaws of the tool as is now. 1. If I pick up the review after a break I don't want to patch the final list of coordinates from different review session. 2. The annotation layer shows all annotations not only for the segment you are currently viewing. If in the last review session you already verfied that all locations you flagged in that session are correct, you may want to visualize only the novel locations. 
Hope I am kind of making this clear here...

Reply:
OK, so is the idea that when load_annotations == False, the seg_error_coordinates are just stored (and appended to as you annotate more), but not actually displayed?

Reply:
Yes, that was my idea behind this because R. had already annotated >100 splits by the time we started this. Large numbers of annotations could be confusing. So if they are not needed because they are all verified from a previous revision round one can leave them out this way... 
(Alternatively one could link the annotation to the errorneous segment of the todo list (requiring batch size to be one). By overriding update_segments one could only display the locations associated with a given batch of segments from the todo list. To view all marked location one could create a "show_all" function. However, as mentioned in the email thread I am not sure whether getting to split locations as implemented here is the most efficient way. Assembling segments belonging to one neurite by selection is usually faster. This would yield more coordinate pairs for a given split, but the would not necessarily be in close proximity to each other. If the latter is what you are after, that should be stated explicitly in the instructions.)